test/api/services/freshtrakApiService.test.ts

```diff
@@ -0,0 +1,46 @@
+import request from 'request-promise';
+import { getFreshTrakEvents } from '../../../api/services/freshtrakApiService';
+import { FRESHTRAK_API_HOST } from '../../../api/utils/constants';
+import freshtrakResponseDto from '../../../api/models/freshtrakAPI/freshtrakResponseDto';
+
+jest.mock('request-promise');
+
+describe('freshtrakApiService', () => {
+  describe('.getFreshTrakEvents', () => {
+    let get = jest.spyOn(request, 'get')
+    beforeEach(() => {
+      get.mockResolvedValue(
+        {
+            agency: 
+            {
+                name: 'Mid-Ohio Foodbank - Kroger Community Pantry',
+                events: [
+                {
+                    name: 'Drive Thru',
+                    exception_note: '',
+                    event_details: 'This is a drive thru distribution. Wear a mask and make sure your trunk is cleaned out. Please, no more than 5 families being served per vehicle, if it fits in your vehicle.',
+
+                },
+
+                ],
+            },
+        });
+    });
+
+    afterEach(() => {
+      get.mockRestore();
+    });
+
+    it('gets a freshtrak agency resource and returns a response object', async () => {
+      const agencyId = 6
+      const url = `${FRESHTRAK_API_HOST}/api/agencies/${agencyId}`;
+      const response = await getFreshTrakEvents(agencyId);
+
+      expect(get).toHaveBeenCalledWith(url);
+      if (response) {
+        expect(response.agency.name).toEqual('Mid-Ohio Foodbank - Kroger Community Pantry');
+        expect(response.agency.events[0].name).toEqual('Drive Thru');
+      }
```

Comment:
```suggestion
      expect(response).toBeTruthy();
      expect(response.agency.name).toEqual('Mid-Ohio Foodbank - Kroger Community Pantry');
      expect(response.agency.events[0].name).toEqual('Drive Thru');
```

Reply:
It doesn't like that one.  I get the error that 'Object is possibly null'

Reply:
You can do `expect(response?.agency.name)`

Reply:
> You can do `expect(response?.agency.name)`
```
expect(response?.agency.events[0].name).toEqual('Drive Thru');
```
Gives an error of 'Expression Expected'.  From what I can gather conditional chaining (?) is shorthand for a tertiary operator that fills in action when the expression is null.  Thoughts?

Reply:
Are you getting that from running the tyepscript compiler or the IDE you're using? I'm not getting that after checking out your branch and trying `npx tsc` and using the optional chaining in that spot.

I would also recommend getting linting setup because I ran lint fix just on that test and there are a bunch of corrections that the linter is making.

Reply:
You're going to have a hard time getting CI to pass until all the lint errors are fixed

Reply:
Scratch that :facepalm: we don't even have the travis ci running the linter! I'm going to put up a PR for that

Reply:
So I am putting up a PR for some fixes with linting and so we have CI run linting before building, this should've been added in the first place I'm guessing we just missed it initially

Reply:
I'm am working through the linter fixes locally.  Very helpful.

Regarding the optional chaining error, I am getting that message in VS code.   If,  I run the  npx tsc it fails with an error:

```
Expected: "Mid-Ohio Foodbank - Kroger Community Pantry"
    Received: undefined

      36 |       //expect(get).toHaveBeenCalledWith(url);
      37 |       //if(response) {
    > 38 |         expect(response?.agency.name).toEqual(
         |                                       ^
      39 |           'Mid-Ohio Foodbank - Kroger Community Pantry'
      40 |         );
      41 |         expect(response?.agency.events[0].name).toEqual('Drive Thru');

      at test/api/services/freshtrakApiService.test.ts:38:39
      at fulfilled (test/api/services/freshtrakApiService.test.ts:5:58)
```
If use the same code within the if(response) block it works fine.

Reply:
That just means that it didn't equal what you were asserting that it did. It's not related to typescript. So the test or the mock needs fixed so that the test passes. The reason it passed before was because it wasn't even getting inside the `if` block because response was undefined which is why I'm saying we should remove the if block.

Reply:
(fyi I got the same error on my end when I ran the test)

Reply:
Ohh.  Sorry about that.   Thanks for putting me on the right track!

Reply:
No problem, happy to help!